onnxscript/test/models/graph_attr.py

```diff
@@ -0,0 +1,34 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from onnxscript import graph, script
+from onnxscript.onnx_opset import opset15 as op
+from onnxscript.onnx_types import BOOL, INT64
+
+
+@script()
+def cumulative_sum(X: INT64['N']):
+    '''Test use of a nested-function as a graph-attribute, using the Scan operator.'''
+    @graph(parent=cumulative_sum)
+    def Sum(sum_in, next):
+        sum_out = sum_in + next
+        scan_out = op.Identity(sum_out)
+        return sum_out, scan_out
+    zero = op.Constant(value_int=0)
+    _, result = op.Scan (zero, X, body=Sum, num_scan_inputs=1)
+    return result
+
+@script()
+def sum_to(X: INT64):
+    '''Test use of a nested-function as a graph-attribute, using the Loop operator.'''
+    @graph(parent=sum_to)
```

Comment:
Yes and no. I hope we can eliminate this requirement eventually, but it is used for now. It is a long explanation. This is because of a combination of features: We'd like the translation of this nested-function to Graph happen once (when sum_to is translated by converter when `@script()(sum_to))` happens, and not when `@graph(LoopBody)` executes (during eager-mode). So, when `@graph` executes in eager-mode we want to lookup and reuse the previously produced GraphProto, which is stored in the IR created for `sum_to`. This is the reason we need access to the parent `sum_to`. We know this parent at conversion-time when `@script` is evaluated on `sum_to`, but it is not easy to get this information when `@graph` is executed during eager-mode. At least, I don't know how to get this information. May be `inspect` will let us get this information? However, the converter (during `@script` time) does have this information and can automatically add this to the AST and re-compile it. But that's a lot of work and not clear we want to do that yet (as we once discussed).

Reply:
Ok, I replaced it by looking it up in the frame.